debian/battery-monitor.1
.\"                                      Hey, EMACS: -*- nroff -*-
.\" (C) Copyright 2022 Himadri Sekhar Basu <hsb10@iitbbs.ac.in>,
.\"
.\" First parameter, NAME, should be all caps
.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
.\" other parameters are allowed: see man(7), man(1)
.TH BATTERY-MONITOR 1 "June 20 2022"
.\" Please adjust this date whenever revising the manpage.
.\"
.\" Some roff macros, for reference:
.\" .nh        disable hyphenation
.\" .hy        enable hyphenation
.\" .ad l      left justify
.\" .ad b      justify to both left and right margins
.\" .nf        disable filling
.\" .fi        enable filling
.\" .br        insert line break
.\" .sp <n>    insert n+1 empty lines
.\" for manpage-specific macros, see man(7)
.SH NAME
battery-monitor \- A utility tool to monitor laptop battery
.SH SYNOPSIS
.B battery-monitor
.RI [ options ]
.br
.SH DESCRIPTION
.PP
.\" TeX users may be more comfortable with the \fB<whatever>\fP and
.\" \fI<whatever>\fP escape sequences to invode bold face and italics,
.\" respectively.
\fBbattery-monitor\fP is a utility tool, notifies user about charging, discharging and not charging state of the laptop battery on Linux.
.SH OPTIONS
battery-monitor launches automatically at start-up as an indicator.
.br
To open the settings window, you may launch battery-monitor with no options.
.TP
.B \-h, \-\-help
Show summary of options.
.TP
.B \-i, \-\-indicator
Start Battery Monitor Indicator.
.TP
.B \-t, \-\-test
Test Battery Monitor for debug purposes when battery is absent.
.TP
.B \-v, \-\-verbose
Print debug messages to stdout i.e. terminal.
.TP
.B \-V, \-\-version
Show version of program.
.SH AUTHORS
battery-monitor was developed by Maksudur Rahman Maateen <maateen@outlook.com> with various contributors
.br
This manual page was originally written by Himadri Sekhar Basu <hsb10@iitbbs.ac.in>.
.SH COPYRIGHT
Copyright (C) 2020-2022 Himadri Sekhar Basu
.br
There is NO warranty. You may redistribute this software under the terms of the GNU General Public License. For more information about these matters, see the files named copyright.


debian/changelog
battery-monitor (1.1.0) stable; urgency=medium

  * fix publish .deb files (#33)
  * Update readme from version 1.0.8 to 1.0.9 (#34)
  * fix no indicator issue and CI failures (#35)
  * Fix threading and remove unused libraries (#36)
    - Use _async to synchronize monitoring thread
    - Fix no icon in indicator for quit option
    - remove lines for unused libraries

 -- Himadri Sekhar Basu <hsb10@iitbbs.ac.in>  Sun, 26 Jun 2022 19:41:40 +0530

battery-monitor (1.0.9) stable; urgency=medium

  * Update readme from version 1.0.7 to 1.0.8-1 (#28)
  * pythonize battery-monitor; closes #29 (#30)
    - use pure python packaging methods
    - update .gitignore
    - update desktop files
    - Parse arguments from cmdline
    - Turn off debug mode by default, use '-v[erbose]'
      to enable it
    - Add options to parse arguments for help, indicator
      and version
    - read version from d/changelog
    - update test module
  * re-debianize to match python packaging (#31)
    - d/compat: deleted
    - d/control: update build and runtime depends
    - d/control: update standards
    - d/rules: use pybuild
    - d/copyright: update
    - d/install: remove 'etc'
    - d/postinst: update
    - d/postrm: add file
    - d/salsa-ci: add
    - d/battery-monitor-doc.docs: add
    - d/battery-monitor.1: add
  *  update workflows (#32)
    - remove old source files
    - Update auto build (CI) workflow
    - Update publish deb packaging workflow
    - d/rules: update autoclean

 -- Himadri Sekhar Basu <hsb10@iitbbs.ac.in>  Mon, 20 Jun 2022 22:40:13 +0530

battery-monitor (1.0.8) stable; urgency=medium

  * Update readme with version 1.0.7 (#26)
  * Add option to use user-defined sound files (#27)
    - Now users can select custom sound files as notification sound
  * Add reset to default button (#21)
    - This helps learning users to restore and use valid values
      in case the config file is corrupted or missing.

 -- Himadri Sekhar Basu <hsb10@iitbbs.ac.in>  Tue, 31 May 2022 20:59:14 +0530

battery-monitor (1.0.7) stable; urgency=medium

  * Update readme with version 1.0.6 (#23)
  * Append random numbers after logfilename (#24)
    - For multiuser systems when a log file exists in
      /tmp, the app cannot recreate the logfile after
      logging into other user
  * create only one log file per user (#25)

 -- Himadri Sekhar Basu <hsb10@iitbbs.ac.in>  Mon, 02 May 2022 23:14:39 +0530

battery-monitor (1.0.6) stable; urgency=medium

  * Update readme with version 1.0.5
  * Use new settings while monitoring; fixes #9 (#22)
    - now new settings will be used within 5 seconds of saving
  * Add and update copyright to 2022 and code cleanup
    - AboutWindow.py: remove unused imports
    - AppIndicator.py: remove unused lines
  * Update CONTRIBUTING and debian compatibility
    - CONTRIBUTING.md: use git repo from @hsbasu
    - debian/compat: Update compatibility to 12
  * Take arguments from command-line 

 -- Himadri Sekhar Basu <hsb10@iitbbs.ac.in>  Tue, 05 Apr 2022 09:40:49 +0530

battery-monitor (1.0.5) stable; urgency=medium

  * add new workflow to publish deb package from latest release (#18)
  * update workflows to run on all branches
  * Update readme with version 1.0.4 and workflows
  * fix libnotify:ERROR while showing success notification; fixes #19
  * processed_battery_info as local variable
  * improve showing state change in notifications loop
  * Enable logging of events for debug, error and info (#20); closes #5
    - Print all log messages on terminal
    - Print info messages to a temporary file
  * modify showing Success Notification (#17)
    - Use switch to enable/disable showing Success Notification
      instead of using an entry
    - Use getboolean to set success and sound switch
    - Change success and sound switch labels based on configuration
  * Add option to modify number of notifications (#16)
    - Now users can choose how many notification to be shown

 -- Himadri Sekhar Basu <hsb10@iitbbs.ac.in>  Thu, 10 Mar 2022 20:59:33 +0530

battery-monitor (1.0.4) stable; urgency=medium

  * Add code quality and CI workflows
  * stop running in the background
  * Update copyright year to 2022
  * Add main menu in Settings Window
  * Add About Window in main menu
  * Add keyboard shortcuts to main menu
  * make save button insensitive post-click
  * Fix no Notification bug (#12)
  * Re-order battery percent variables list
    - write battery percent variables in decreasing order
      starting with upper warning and ending at critical battery
  * Update ui file to include new features
    - Add notification count entry
    - Add reset to default button
    - Add enable/disable sound switch
    - Modify a few labels and tooltips
  * Add option to enable/disable sound notification; closes #4 (#15)

 -- Himadri Sekhar Basu <hsb10@iitbbs.ac.in>  Sun, 06 Mar 2022 10:52:08 +0530

battery-monitor (1.0.3) stable; urgency=medium

  * Update readme
  * Add ui file
  * Update debian compatibility
  * Add schema for battery-monitor
  * Update ui with new features
  * Update settings window to match new ui; closes #8

 -- Himadri Sekhar Basu <hsb10@iitbbs.ac.in>  Mon, 28 Feb 2022 10:56:01 +0530

battery-monitor (1.0.2) stable; urgency=medium

  * Silence success notifiaction at user's choice; fixes #2 (#11)

 -- Himadri Sekhar Basu <hsb10@iitbbs.ac.in>  Sun, 07 Nov 2021 23:19:08 +0530

battery-monitor (1.0.1) stable; urgency=medium

  * Update .desktop files via generate_desktop_files (#9)
  * Show charging/discharging Notification; fixes #5 (#8)
  * Change source compatibility format to 10
  * Quit if acpi is not installed or battery is not present

 -- Himadri Sekhar Basu <hsb10@iitbbs.ac.in>  Sat, 30 Oct 2021 21:11:18 +0530

battery-monitor (1.0.0) stable; urgency=medium

  * Setup translation using gettext and update about dialog
  * Shift to debian packaging from python.
  * Added support for playing sound.
  * Feature: Upper threshold warning
  * Made a svg file for project icon

usr/lib/battery-monitor/battery-monitor.py
#!/usr/bin/python3

# Copyright: 2021-2022 Himadri Sekhar Basu <hsb10@iitbbs.ac.in>
#
# This file is part of battery-monitor.
#
# battery-monitor is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# battery-monitor is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with battery-monitor. If not, see <http://www.gnu.org/licenses/>
# or write to the Free Software Foundation, Inc., 51 Franklin Street,
# Fifth Floor, Boston, MA 02110-1301, USA..
#
# Author: Himadri Sekhar Basu <hsb10@iitbbs.ac.in>
#

# standard library
import getpass
import glob
import logging
import setproctitle
import signal
import string
import sys
from random import choice

# third-party library
import gi
gi.require_version('Gtk', '3.0')
from gi.repository import Gtk

# imports from current project
from AppIndicator import AppIndicator, bm_daemon

setproctitle.setproctitle("battery-monitor")

## Setup logging
def create_logfile():
	username = getpass.getuser()
	random_code =  ''.join(choice(string.digits) for _ in range(4))
	if len(glob.glob('/tmp/battery-monitor_'+username+'*')) ==0:
		logfile = '/tmp/battery-monitor_' + username + '_' + random_code + '.log'
	else:
		logfile = glob.glob('/tmp/battery-monitor_'+username+'*')[0]
	
	return logfile

# Create logger
logger = logging.getLogger('Battery Monitor')
# Set logging level
logger.setLevel(logging.DEBUG)

# Create StreamHandler which logs even debug messages
cHandler = logging.StreamHandler()
# Set level for StreamHandler
cHandler.setLevel(logging.DEBUG)

# create file handler which logs only info messages
# Set the log filename
logfile = create_logfile()
fHandler = logging.FileHandler(logfile)
# Set level for FileHandler
fHandler.setLevel(logging.INFO)

# create formatter and add it to the handlers
log_format = logging.Formatter('%(asctime)s %(name)s - %(levelname)s: %(message)s')
cHandler.setFormatter(log_format)
fHandler.setFormatter(log_format)

# add the handlers to the logger
logger.addHandler(cHandler)
logger.addHandler(fHandler)


def main() -> None:
	# checking Test Mode enabled or not
	try:
		if sys.argv[1] == '--test':
			TEST_MODE = True
		else:
			TEST_MODE = False
	except IndexError:
		TEST_MODE = False
	
	# initiaing app indicator
	logger.info("Battery Monitor Started.")
	AppIndicator()
	bm_daemon(TEST_MODE)
	Gtk.main()

if __name__ == '__main__':
	signal.signal(signal.SIGINT, signal.SIG_DFL)
	main()
